istioctl/cmd/auth.go

```diff
@@ -268,6 +269,8 @@ func init() {
 		"Authorization policy files")
 	upgradeCmd.PersistentFlags().StringSliceVarP(&serviceFiles, "service", "s", []string{},
 		"Kubernetes Service resource that provides the mapping relationship between service name and pod labels")
+	upgradeCmd.PersistentFlags().StringVarP(&meshConfig, "mesh-config", "m", "",
```

Comment:
Could you use `meshConfigFile` be consistent with the flag used in `kube-inject`? Also add a description for the flag.

Reply:
If you mean changing the name to `meshConfigFile`, it's not possible since it's already been declared in the `cmd` package (kube-inject declares it). If you mean using `meshConfigFile` from (kube-inject), we can move it out of the kube-inject file to the root file.

Reply:
I mean to change `"mesh-config"` to `"meshConfigFile"` so that user could use the same flag for the same thing in both commands. Do you mean this is impossible? Hmm, that is surprising.

Reply:
Both auth.go and kube-inject.go are under the cmd package, so we can either use a different name for auth.go, or move it out of kube-inject to make it look better (so not making it kube-inject only) and share the same variable.

Reply:
I don't care about the variable name because it's a pure internal implementation detail, also I think we don't need to share the same variable because they are used in different sub commands, we could but I don't feel there is any special benefit to do so.

Here I'm asking for changing the **flag name** to `meshConfigFile` which is the 2nd argument. This makes it consistent with the existing flag used in the `kube-inject` command: https://github.com/istio/istio/blob/7774d862b5d93e616523ff3219294d8e49d00ea2/istioctl/cmd/kubeinject.go#L319

This gives the consistent experience for customers so that they can use the same flag for the same thing through out istioctl, even these are in different sub-commands. Also could you use which could be just `mesh configuration filename` here.

Reply:
Okay I thought you meant the variable. I'll change the flag name.